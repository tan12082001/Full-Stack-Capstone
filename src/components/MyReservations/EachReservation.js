import PropTypes from 'prop-types';
import React from 'react';

const EachReservation = ({ reservation }) => (
  <div className="each-reservation-div" key={reservation.id}>
    <span>{`Booked For 🚗: ${reservation.car && reservation.car.name ? reservation.car.name : 'Meow'}`}</span>
    <span>{`Reservation is on 📆: ${reservation.date}`}</span>
    <span>{`Reservation is at  📍: ${reservation.city}`}</span>
  </div>
);

EachReservation.propTypes = {
  reservation: PropTypes.shape({
    id: PropTypes.number.isRequired,
    car: PropTypes.shape({
      name: PropTypes.string.isRequired,
    }).isRequired,
    date: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
  }).isRequired,
};

export default EachReservation;
