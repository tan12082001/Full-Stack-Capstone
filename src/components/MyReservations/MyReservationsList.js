import React from 'react';
import { useSelector } from 'react-redux';

const MyReservationsList = () => {
  const myReservations = useSelector((state) => state.reservation.reservations);
  const status = useSelector((state) => state.reservation.status);
  const error = useSelector((state) => state.reservation.error);

  if (status === 'loading') {
    return <p>Loading...</p>;
  }

  if (status === 'failed') {
    return (
      <p>
        Error:
        {error}
      </p>
    );
  }

  return (
    <div className="my-reservations-inner">
      {myReservations && myReservations.map((reservation) => (
        <div key={reservation.id} className="each-reservation-div">
          <span>{`Car: ${reservation.car.name ? reservation.car.name : 'non nome'}`}</span>
          <span>{`Date: ${reservation.date}`}</span>
          <span>{`City: ${reservation.city}`}</span>
        </div>
      ))}
    </div>
  );
};

export default MyReservationsList;
