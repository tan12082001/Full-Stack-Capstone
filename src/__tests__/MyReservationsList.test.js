import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import MyReservations from '../pages/LandingPage/MyReservation/MyReservation';

const mockStore = configureMockStore();

describe('MyReservations Component', () => {
  it('renders reservations present', () => {
    const store = mockStore({
      authentication: {
        authenticatedUser: {
          username: 'testuser',
        },
        status: 'succeeded',
      },
      reservation: {
        reservations: [
          {
            date: '2023-10-10',
            city: 'City A',
          },
          {
            date: '2022-11-11',
            city: 'City B',
          },
        ],
      },
    });

    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <MyReservations />
        </MemoryRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
