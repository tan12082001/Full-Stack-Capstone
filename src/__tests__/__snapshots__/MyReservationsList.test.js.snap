// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyReservations Component renders reservations present 1`] = `
<div>
  <div
    class="my-reservations-page-outer"
  >
    <h1>
      My Current Reservations
    </h1>
    <div
      class="my-reservations-inner"
    >
      <h1>
        My Reservations
      </h1>
      <div
        class="each-reservation-div"
      >
        <span>
          Car: undefined
        </span>
        <span>
          Date: 2023-10-10
        </span>
        <span>
          City: City A
        </span>
      </div>
      <div
        class="each-reservation-div"
      >
        <span>
          Car: undefined
        </span>
        <span>
          Date: 2022-11-11
        </span>
        <span>
          City: City B
        </span>
      </div>
    </div>
  </div>
</div>
`;
